### DISCUSSION TITLE
--inspect[=[host:]port]

### DISCUSSION THREAD

**Discussion by Guid75** [vitest-dev/vitest#4062] [Ideas]
Hi,

Wondering why vitest does not allow to pass host and port to --inspect and --inspect-brk flags like nodejs does.

My process is running on a docker container and I need to pass "0.0.0.0" instead to "127.0.0.1" in order to access it from my debugger.

**Comment by AriPerkkio** [comment]
Vitest uses `node:inspector` APIs to set up debugger.

https://github.com/vitest-dev/vitest/blob/0c13c39c2ddc5cd3deea0636d64f24d95fc93bdd/packages/vitest/src/runtime/inspector.ts#L8-L21

Currently `--inspect` and `--inspect-brk` are boolean flags. I think your use case would be covered if it allowed URL as argument as well. That could be passed to `inspect.open()` as argument. https://nodejs.org/dist/latest-v18.x/docs/api/inspector.html#inspectoropenport-host-wait

**Comment by jhawson-lmi** [comment]
I have the same situation as @Guid75; my test process runs in a docker container. It would be incredibly helpful to be able to debug it in that environment. 

As @AriPerkkio pointed out, the node `inspector.open()` method provides the necessary flexibility. Specifically, it accepts `port` and `host` arguments: 
```
inspector.open([port[, host[, wait]]])
```
These arguments determine the port and host to which the inspector will bind. By default, the host is `localhost`, which makes sense for most local development scenarios. But in a container, that makes it difficult to expose the port for incoming connections (requires host networking mode, or the use of tools like socat to port-forward; these are not great options). 

With these arguments, vitest could implement the same command-line syntax as [node itself](https://nodejs.org/en/learn/getting-started/debugging#command-line-options).

**Reply by AriPerkkio** [reply]
There's now feature request for this at https://github.com/vitest-dev/vitest/issues/5500.
